js/web/lib/wasm/jsep/webgpu/gpu-data-manager.ts

```diff
@@ -312,20 +326,38 @@ class GpuDataManagerImpl implements GpuDataManager {
       buffer.destroy();
     }
     this.buffersForUploadingPending = [];
-    for (const buffer of this.buffersPending) {
-      // eslint-disable-next-line no-bitwise
-      if ((buffer.usage & GPUBufferUsage.STORAGE) === GPUBufferUsage.STORAGE) {
-        // Put the pending buffer to freeBuffers list instead of really destroying it for buffer reusing.
-        this.freeBuffers.get(buffer.size)!.push(buffer);
+
+    if (this.buffersPending.length === 0) {
+      return;
+    }
+
+    // Don't release intermediate tensors in non-default mode.
```

Comment:
(**MARK** for my self. I probably need to think more carefully about this part.)

in theory, some intermediate result can be "released"(not destroy) and used later by a different kernel, if its size fits the requirement.

Reply:
You are right. Change it to reuse storage buffers and only keep uniform buffers unreleased.

Reply:
I just reverted the changes to reuse storage buffer. I missed to consider the situation that if we have multiple captured graphs. It may result incorrect result if they reuse the same storage buffers (one graph will override the other graph's data). For graph mode, we may reuse the storage buffer for the current captured graph and not share with other models. Will do it in a separate PR once this PR is merged.